packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -18,63 +17,122 @@ describe('SyntheticEvent', () => {
   var createEvent;
 
   beforeEach(() => {
-    // TODO: can we express this test with only public API?
-    SyntheticEvent = require('SyntheticEvent');
     React = require('react');
     ReactDOM = require('react-dom');
     ReactTestUtils = require('react-dom/test-utils');
-
-    createEvent = function(nativeEvent) {
-      var target = require('getEventTarget')(nativeEvent);
-      return SyntheticEvent.getPooled({}, '', nativeEvent, target);
+    createEvent = (eventInit, eventOptions) => {
+      const defaultEventInit = {
+        bubbles: true,
+        cancelable: true,
+        ...eventOptions,
+      };
+      var event = new Event(eventInit, defaultEventInit);
+      event.initEvent(
+        eventInit,
+        defaultEventInit.bubbles,
+        defaultEventInit.cancelable,
+      );
+      return event;
     };
   });
 
   it('should normalize `target` from the nativeEvent', () => {
-    var target = document.createElement('div');
-    var syntheticEvent = createEvent({srcElement: target});
```

Comment:
In my https://github.com/facebook/react/pull/11332#discussion_r147254081 comment, I didn't ask to remove `srcElement` from *all* tests. I was saying that only one test intentionally tested for `srcElement` support (that's why the test existed). That's why after refactoring, there should *also * be one test that verifies this.